hail/python/hail/experimental/import_gtf.py

```diff
@@ -160,3 +162,86 @@ def import_gtf(path, reference_genome=None, skip_invalid_contigs=False, min_part
+       -------
+       >>> hl.filter_intervals(ht, get_gene_intervals(gene_symbols=['PCSK9']))  # doctest: +SKIP
+
+       Parameters
+       ----------
+
+       gene_symbols : :obj:`list` of :obj:`str`, optional
+           Gene symbols (e.g. PCSK9).
+       gene_ids : :obj:`list` of :obj:`str`, optional
+           Gene IDs (e.g. ENSG00000223972).
+       transcript_ids : :obj:`list` of :obj:`str`, optional
+           Transcript IDs (e.g. ENSG00000223972).
+       verbose : :obj:`bool`
+           If ``True``, print which genes and transcripts were matched in the GTF file.
+       reference_genome : :obj:`str` or :class:`.ReferenceGenome`, optional
+           Reference genome to use (passed along to import_gtf).
+       gtf_file : :obj:`str`
+           GTF file to load. If none is provided, but `reference_genome` is one of
+           `GRCh37` or `GRCh38`, a default will be used.
+
+       Returns
+       -------
+       :obj:`list` of :class:`.Interval`
+    """
+    GTFS = {
+        'GRCh37': 'gs://hail-common/references/gencode/gencode.v19.annotation.gtf.bgz',
+        'GRCh38': 'gs://hail-common/references/gencode/gencode.v29.annotation.gtf.bgz',
+    }
+    if reference_genome is None:
+        reference_genome = hl.default_reference().name
+    if gtf_file is None:
+        gtf_file = GTFS.get(reference_genome)
+        if gtf_file is None:
+            raise ValueError('get_gene_intervals requires a GTF file, or the reference genome be one of GRCh37 or GRCh38')
+    if not gene_symbols and not gene_ids and not transcript_ids:
+        raise ValueError('get_gene_intervals requires at least one of gene_symbols, gene_ids, or transcript_ids')
+    ht = hl.experimental.import_gtf(gtf_file, reference_genome=reference_genome,
+                                    skip_invalid_contigs=True, min_partitions=12)
+    ht = ht.annotate(gene_id=ht.gene_id.split(f'\\.')[0],
+                     transcript_id=ht.transcript_id.split('\\.')[0])
+    criteria = [lambda x: False]
+    if gene_symbols:
+        criteria.append(lambda x: hl.any(lambda y: (x.feature == 'gene') & (x.gene_name == y), gene_symbols))
+    if gene_ids:
+        criteria.append(lambda x: hl.any(lambda y: (x.feature == 'gene') & (x.gene_id == y.split('\\.')[0]), gene_ids))
+    if transcript_ids:
+        criteria.append(lambda x: hl.any(lambda y: (x.feature == 'transcript') & (x.transcript_id == y.split('\\.')[0]), transcript_ids))
+
+    def combine_functions(func_list, x, operator='and'):
+        possible_operators = ('and', 'or')
+        if operator not in possible_operators:
+            raise ValueError(f'combine_functions only allows operators: {", ".join(possible_operators)}')
+        cond = func_list[0](x)
+        for c in func_list[1:]:
+            if operator == 'and':
+                cond &= c(x)
+            elif operator == 'or':
+                cond |= c(x)
+        return cond
+
+    ht = ht.filter(combine_functions(criteria, ht, operator='or'))
```

Comment:
you can actually rewrite this as:

```
import operator
import functools

functools.reduce(operator.ior, criteria)
```

Reply:
er, the bit with the lambda makes it more complicated. Do you need to defer those computations?  

Can it be:

```
    if transcript_ids:
        criteria.append(hl.any(lambda y: (ht.feature == 'transcript') & (ht.transcript_id == y.split('\\.')[0]), transcript_ids))
```